src/PortingAssistant.Client.Common/Utils/FileSystemAccess.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.IO;
+using System.Security.Principal;
+using System.Security.AccessControl;
+
+namespace PortingAssistant.Client.Common.Utils
+{
+    public static class FileSystemAccess
+    {
+        /// <summary>
+        /// Checks directory and all content for write access
+        /// </summary>
+        /// <param name="path">Directory path</param>
+        /// <returns>First file/directory with no write access found if any, blank if all contents have access</returns>
+        public static string CheckWriteAccessForDirectory(string path)
```

Comment:
should we return all inaccessible files or dirs instead of  the first one ?

Reply:
@flywanli made the change here: https://github.com/aws/porting-assistant-dotnet-client/pull/103/commits/a3c4609f52d2799c352b5834f0936f3c64c0cede